### ISSUE TITLE
[Material] VisualStateManager does not work properly on the Entry TextColor property

### ISSUE THREAD

**Issue by SamuelLioult** [xamarin/Xamarin.Forms#5732]
### Description

The Visual State Manager does not change TextColor properly on an Entry on Android with Material visual.
Normally, with Material, the TextColor affects the text color (obvious) and the bottom line.
But, it seems there is a bug because the bottom line has never the same color value of the text.
So when the text has the focused color, the bottom line has the normal color and vice versa.

### Steps to Reproduce

1. Set the Entry style:
```xaml
<Style TargetType="Entry">
    <Setter Property="BackgroundColor" Value="Transparent"/>
    <Setter Property="VisualStateManager.VisualStateGroups">
        <VisualStateGroupList>
            <VisualStateGroup x:Name="CommonStates">
                <VisualState x:Name="Normal">
                    <VisualState.Setters>
                        <Setter Property="TextColor" Value="Default"/>
                    </VisualState.Setters>
                </VisualState>
                <VisualState x:Name="Focused">
                    <VisualState.Setters>
                        <Setter Property="TextColor" Value="{StaticResource PrimaryColor}"/>
                    </VisualState.Setters>
                </VisualState>
            </VisualStateGroup>
        </VisualStateGroupList>
    </Setter>
</Style>
```
2. Add several entries in your page
3. Change focused Entry

### Expected Behavior

The bottom line and the text have the same color, according to their state.

### Actual Behavior

The text has the good color, but the bottom line has the opposite color.

### Basic Information

- Version with issue: Xamarin Forms 3.6.0.264807
- Last known good version: No history
- IDE: Visual Studio 2017 15.9.9
- Platform Target Frameworks: 
  - Android: 9.0
- Android Support Library Version: v7
- Affected Devices: Nokia 6.1

### Screenshots
![screenshot](https://user-images.githubusercontent.com/10460974/55163545-45be6680-516a-11e9-8649-dbdc000ae25f.png)
![screenshot2](https://user-images.githubusercontent.com/10460974/55163705-8d44f280-516a-11e9-868f-73bd6b76b139.png)

**Comment by PureWeen** [comment]
duplicate https://github.com/xamarin/Xamarin.Forms/issues/5714
